### PR TITLE
Added a PyCharm Diff reporter.

### PR DESCRIPTION
**Pull request by emilybache**
It is nearly the same as a GenericDiffReporter but needs an extra argument. I created a subclass and added it so that it will be constructed in GenericDiffReporterFactory.

I'd like someone more familiar with the codebase to go through these changes and see if it is in keeping with the rest of the design

### CODE REVIEW

**Comment by tdpreece** [comment]
Hi,

Thanks for the PR. Passing options to a diffing tool seems like something that will come up again in the future.

I wonder if we could handle this in a more general way. What do you think about something like the following? (I've not run or tested this)

```
diff --git a/approvaltests/reporters/generic_diff_reporter.py b/approvaltests/reporters/generic_diff_reporter.py
index 9b612e2..cc9836d 100644
--- a/approvaltests/reporters/generic_diff_reporter.py
+++ b/approvaltests/reporters/generic_diff_reporter.py
@@ -14,6 +14,7 @@ class GenericDiffReporter(Reporter):
     def __init__(self, config):
         self.name = config[0]
         self.path = config[1]
+        self.options = config[2] if len(config) == 3 else []
 
     def __str__(self):
         return json.dumps(
@@ -35,11 +36,7 @@ class GenericDiffReporter(Reporter):
         subprocess.call(command_array)
 
     def get_command(self, received, approved):
-        return [
-            self.path,
-            received,
-            approved
-        ]
+        return [self.path] + self.options + [received, approved]
 
     def report(self, received_path, approved_path):
         if not self.is_working:
diff --git a/approvaltests/reporters/reporters.json b/approvaltests/reporters/reporters.json
index d43efaa..c48397c 100644
--- a/approvaltests/reporters/reporters.json
+++ b/approvaltests/reporters/reporters.json
@@ -46,5 +46,10 @@
   [
     "DiffMerge",
     "/Applications/DiffMerge.app/Contents/MacOS/DiffMerge"
+  ],
+  [
+    "PyCharm",
+    "/Applications/PyCharm CE.app/Contents/MacOS/pycharm",
+    ["diff"],
   ]
 ]
```

**Comment by emilybache** [comment]
That works too. I updated the pull request. Do you think it looks better now?

**Comment by tdpreece** [comment]
Hi,

Yes, I prefer this solution. 

Sorting the keys in `approvaltests.reporters.generic_diff_reporter.GenericDiffReporter#__str__` should fix the tests in Python2 this.

```
diff --git a/approvaltests/reporters/generic_diff_reporter.py b/approvaltests/reporters/generic_diff_reporter.py
index cd5d10e..67f9c72 100644
--- a/approvaltests/reporters/generic_diff_reporter.py
+++ b/approvaltests/reporters/generic_diff_reporter.py
@@ -29,7 +29,8 @@ class GenericDiffReporter(Reporter):
         return json.dumps(
             config,
             indent=4,
-            separators=(',', ': ')
+            separators=(',', ': '),
+            sort_keys=True,
         )
 
     @staticmethod
diff --git a/tests/approved_files/GenericDiffReporterTests.test_get_pycharm_reporter.approved.txt b/tests/approved_files/GenericDiffReporterTests.test_get_pycharm_reporter.approved.txt
index d7a61cc..882a89c 100644
--- a/tests/approved_files/GenericDiffReporterTests.test_get_pycharm_reporter.approved.txt
+++ b/tests/approved_files/GenericDiffReporterTests.test_get_pycharm_reporter.approved.txt
@@ -1,7 +1,7 @@
 {
-    "name": "PyCharm",
-    "path": "/Applications/PyCharm CE.app/Contents/MacOS/pycharm",
     "arguments": [
         "diff"
-    ]
+    ],
+    "name": "PyCharm",
+    "path": "/Applications/PyCharm CE.app/Contents/MacOS/pycharm"
 }
\ No newline at end of file
```

File: tests/test_genericdiffreporter.py [hunk 1]
```diff
@@ -163,3 +163,6 @@ def assertFileIsEmpty(self, file_path):
         if file_size != 0:
             msg = "File is not empty: {}" % file_path
             self.fail(msg)
+
```
Comment (tdpreece) [inline tests/test_genericdiffreporter.py]: Tiny suggestion: I think it would be nice to keep this test with the other reports (llines 27-38).

**Comment by tdpreece** [comment]
Once the tests are passing I'm happy to merge this. Thanks.

**Comment by emilybache** [comment]
sorry about the failing tests. I didn't even have python2 installed on my system and hadn't realized the key sorting was needed for that. Hope it looks ok now?

**Comment by tdpreece** [comment]
Looks good. Thanks!

**Pull request merged**